src/main/catalog-sources/kubeconfig-sync.ts

```diff
@@ -258,37 +253,59 @@ function diffChangedConfig(filePath: string, source: RootSource): Disposer {
   return cleanup;
 }
 
-async function watchFileChanges(filePath: string): Promise<[IComputedValue<CatalogEntity[]>, Disposer]> {
-  const stat = await fse.stat(filePath); // traverses symlinks, is a race condition
-  const watcher = watch(filePath, {
-    followSymlinks: true,
-    depth: stat.isDirectory() ? 0 : 1, // DIRs works with 0 but files need 1 (bug: https://github.com/paulmillr/chokidar/issues/1095)
-    disableGlobbing: true,
-    ignorePermissionErrors: true,
-    usePolling: false,
-    awaitWriteFinish: {
-      pollInterval: 100,
-      stabilityThreshold: 1000,
-    },
-  });
+function watchFileChanges(filePath: string): [IComputedValue<CatalogEntity[]>, Disposer] {
   const rootSource = new ExtendedObservableMap<string, ObservableMap<string, RootSourceValue>>();
   const derivedSource = computed(() => Array.from(iter.flatMap(rootSource.values(), from => iter.map(from.values(), child => child[1]))));
-  const stoppers = new Map<string, Disposer>();
 
-  watcher
-    .on("change", (childFilePath) => {
-      stoppers.get(childFilePath)();
-      stoppers.set(childFilePath, diffChangedConfig(childFilePath, rootSource.getOrInsert(childFilePath, observable.map)));
-    })
-    .on("add", (childFilePath) => {
-      stoppers.set(childFilePath, diffChangedConfig(childFilePath, rootSource.getOrInsert(childFilePath, observable.map)));
-    })
-    .on("unlink", (childFilePath) => {
-      stoppers.get(childFilePath)();
-      stoppers.delete(childFilePath);
-      rootSource.delete(childFilePath);
-    })
-    .on("error", error => logger.error(`${logPrefix} watching file/folder failed: ${error}`, { filePath }));
+  let watcher: FSWatcher;
+
+  (async () => {
```

Comment:
How about implementing this as `watch` method?

Reply:
That doesn't work, none of the OS's or the libraries that I tried reported any events when watching a path that doesn't exist at the time of call.